src/Rasa.Utils/Networking/LengthedSocket.cs

```diff
@@ -161,10 +164,12 @@ private void OperationCompleted(object o, SocketAsyncEventArgs args)
                         break;
 
                     case SocketAsyncOperation.Receive:
+                        // This value may change in the middle of processing, causing odd behavior
```

Comment:
Please only include the multiple receive fix in this PR, and we can take a look at the fragmented receive in another one, when the code is done.

Reply:
@Blumster Reversed the fragmented receive logic, as requested.